src/main/java/net/inpercima/runandfun/service/RunAndFunServiceImpl.java

```diff
@@ -135,13 +135,14 @@ private RunkeeperActivities getActivities(final String accessToken, final LocalD
     @Override
     public int indexActivities(final String accessToken) {
         final Collection<AppActivity> activities = new ArrayList<>();
-
         final String username = getAppState(accessToken).getUsername();
         for (final RunkeeperActivityItem item : getActivities(accessToken, calculateFetchDate()).getItemsAsList()) {
-            final AppActivity activity = new AppActivity(item.getId(), username, item.getType(), item.getDate(),
-                    item.getDistance(), item.getFormattedDuration());
-            LOGGER.debug("prepare {}", activity);
-            activities.add(activity);
+            if(!repository.exists(item.getId())) {
```

Comment:
Sounds good. Maybe you could refactor the whole for-loop into a filtered stream?